File: 1_C/mario/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // prompt user for an int between 1 and 8 inclusive
    int n;
    do
    {
        n = get_int("height: ");
    }
    while (n < 1 || n > 8);

    // print the pyramid
    for (int i = 0; i < n; i++)
    {
        // print left pyramid
        for (int j = 0; j < n; j++)
        {
            if (i + j < n - 1)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }

        // print margin
        for (int j = n; j == n || j == n + 1; j++)
        {
            printf(" ");
        }

        // print right pyramid
        for (int j = n; j < n + 1 + i; j++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: 6_python/pset6/mario/more/mario.py
n = input("Height: ")
while ((n > 1) and (n < 8)):
    try:
        n = int(input("Height: "))
    except:
        continue

i = 0
while i < n:

    # left
    j = 0
    while j < n:
        if (i + j < n - 1):
            print(' ')
        else:
            print('#')
        j += 1

    # margin
    j = n
    while j == n:
        print(' ')
        j += 1

    # right
    j = n
    while j < n + 1 + i:
        print('#')
        j += 1


    i += 1





